local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,340 @@
+"""Describes obstacles which the Sailbot must avoid: Boats and Land"""
+
+import math
+
+import numpy as np
+from custom_interfaces.msg import HelperAISShip
+from numpy.typing import NDArray
+from shapely.geometry import Point, Polygon
+
+from local_pathfinding.coord_systems import (
+    XY,
+    LatLon,
+    knots_to_kilometers_per_hour,
+    latlon_to_xy,
+    meters_to_km,
+)
+
+# Constants
+PROJ_TIME_NO_COLLISION = 3  # hours
+COLLISION_ZONE_SAFETY_BUFFER = 0.5  # km
+
+
+class Obstacle:
+
+    """
+    This Interface describes general obstacle objects which are
+    anything which the sailbot must avoid.
+
+    The choice was made to create an interface rather than a class as Boat and LandMass objects
+    may be very different objects which share an "avoidable" property and a position, but not much
+    else.
+
+    """
+
+    def __init__(
+        self,
+        reference: LatLon,
+        sailbot_position: LatLon,
+        sailbot_speed: float,
+    ):
+        """
+        Args:
+            -reference (LatLon): latitude and longitude of the next global waypoint
+            -sailbot_position (LatLon): lat and lon position of SailBot
+            -sailbot_speed (float): speed of the SailBot in km/h
+        """
+        self.reference = reference
+
+        # Position of the Sailbot in XY
+        self.sailbot_position = latlon_to_xy(reference, sailbot_position)
+
+        self.sailbot_speed = sailbot_speed
+
+        # This should be defined in child classes
+        self.collision_zone = Polygon(None, None)
```

Comment:
What do you think about this for debugging purposes
- Set `self.collision_zone = None` here
- In `is_valid()`
  ```py
  if self.collision_zone is None:
    raise ValueError("Collision zone has not been initialized")
  ```

Reply:
If you implement this, you will probably want to add the corresponding test

Reply:
I like this idea, I will implement it.

Reply:
You will also want to update the docstring

```py
"""

Raises:
    ValueError: When `self.collision_zone` has not been initialized.
"""
```